joomla/joomla-cms
# Update to TinyMCE 4.0.5

This PR updates TinyMCE editor to version 4.0.5

http://joomlacode.org/gf/project/joomla/tracker/?action=TrackerItemEdit&tracker_item_id=31389
## Screenshots

Simple View:
![image](https://f.cloud.github.com/assets/1986000/758804/0a3cd15e-e759-11e2-88f7-c5a38b35cde3.png)

Advanced View:
![image](https://f.cloud.github.com/assets/1986000/758808/28c1a546-e759-11e2-8bc1-001a6293b2e1.png)

Extended View:
![image](https://f.cloud.github.com/assets/1986000/764437/577f8a34-e811-11e2-9b05-c3ec25f50798.png)
## Opinions wanted

These can be discussed here: https://groups.google.com/forum/?fromgroups=#!topic/joomla-dev-cms/_WG3lRCWhwM
- There don't appear to be any skins for TinyMCE4 yet - this might have implications on Hathor + other admin templates  (Talk to UX team) UPDATE: Post: http://ux.joomla.org/forum/Pick-a-task-waiting-for-you-here/1670-tinymce-4#1679
- Investigate adding more params for the extended view buttons
- Investigate an alternate set-up that uses menu only rather than buttons
- Reinvestigate what buttons we want to show by default in each view
- See note in section below about the Layers plugin - does anyone know anything about this?
## Form Fields Removed due to Plugin removals

Plugin Removals: Plugins that have been removed going from version 3->4 are labelled here: http://www.tinymce.com/wiki.php/Tutorial:Migration_guide_from_3.x
- Format Date, Insert Time, Format Time (Now all included within the Insert Date param)
- Advanced Image (Was replaced by alternative plugin - need to investigate alternative param)
- Advanced Link (Was replaced by alternative plugin - need to investigate alternative param)
- XHTML Extras (Removed by TinyMCE)
- Layers plugin (buttons don't appear to work anymore - does anyone know anything about this)
## Other Changes
- Simple/Advanced themes have been dropped
- Mobile parameter (by default off) enlarges the buttons and uses the simple functionality on mobile devices.
## Testers
- Check if plugin(-xtd) buttons work if there is more than one instance of the editor on the page due to the way the code had to be edited (install a more complex button from JED if possible and check that this still works fine)
- Check the params used in the extended view remove both the menu and buttons!
- Check that the fields work in both front-end, back-end and in Hathor.
- Check the mobile view functions as expected with the larger icons and simplified view as well as turning off the mobile param and checking all buttons work as on the desktop
- Anything else you can think of that is related to the editor
## Known Issues
- The way templates have been included has changed from TinyMCE 3 to 4 so these are now included in the initialization code for TinyMCE and not by an external file! I have added in some compatiblity code for this. Note in Joomla 4.x we'll want to amend the admin script to remove the template_list.js file. I'm going to replace this with repeatable form elements. But this will come in a separate PR.
